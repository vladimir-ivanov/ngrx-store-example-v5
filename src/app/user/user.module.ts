import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { StoreModule } from '@ngrx/store';
import * as fromUser from './reducers/user.reducer';
import { EffectsModule } from '@ngrx/effects';
import { UserEffects } from './effects/user.effects';
import { UserComponent } from './user.component';

@NgModule({
  declarations: [UserComponent],
  imports: [
    CommonModule,
    StoreModule.forFeature('user', fromUser.reducer),
    EffectsModule.forFeature([UserEffects])
  ],
  exports: [
    UserComponent
  ]
})
export class UserModule { }
