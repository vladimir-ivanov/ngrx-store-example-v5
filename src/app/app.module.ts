import {BrowserModule} from '@angular/platform-browser';
import {NgModule} from '@angular/core';

import {AppRoutingModule} from './app-routing.module';
import {AppComponent} from './app.component';
import {StoreModule} from '@ngrx/store';
import {metaReducers, reducers} from './reducers';
import {StoreDevtoolsModule} from '@ngrx/store-devtools';
import {environment} from '../environments/environment';
import {UserModule} from './user/user.module';
import {EffectsModule} from '@ngrx/effects';

@NgModule({
  declarations: [
    AppComponent
  ],
  imports: [
    BrowserModule,
    AppRoutingModule,
    UserModule,
    StoreModule.forRoot(reducers, { metaReducers }),
    EffectsModule.forRoot([]),
    !environment.production ? StoreDevtoolsModule.instrument() : []
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
