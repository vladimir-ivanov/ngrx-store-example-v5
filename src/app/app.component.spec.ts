import {TestBed, async} from '@angular/core/testing';
import {RouterTestingModule} from '@angular/router/testing';
import {AppComponent} from './app.component';
import {UserModule} from './user/user.module';
import {UserComponent} from './user/user.component';

describe('AppComponent', () => {
  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent, UserComponent
      ],
    }).compileComponents();
  }));

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.componentInstance;
    expect(app).toMatchSnapshot();
  });

  it(`should have as title 'ngrx-store-example-v5'`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.debugElement.nativeElement;
    expect(app).toMatchSnapshot();
  });
});
