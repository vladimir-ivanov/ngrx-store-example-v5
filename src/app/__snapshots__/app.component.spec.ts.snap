// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should create the app 1`] = `
AppComponent {
  "title": "ngrx-store-example-v5",
}
`;

exports[`AppComponent should have as title 'ngrx-store-example-v5' 1`] = `
<div
  id="root1"
  ng-version="7.0.2"
>
  <div
    style="text-align:center"
  >
    <h1>
       Welcome to 
    </h1>
  </div>
  <app-user>
    <p>
       user works!

    </p>
  </app-user>
  <router-outlet />
</div>
`;
